### DISCUSSION TITLE
I need help with scoring region parameterization

### DISCUSSION THREAD

**Discussion by ProfLeao** [nrc-cnrc/EGSnrc#744] [Questions] [answered]
```
:start scoring options:
    :start calculation geometry:
        geometry name          = SimulatedUniverse
        scoring regions        = ??? (A)
        scoring region masses  = ??? (B)
        FD geometry            = Phantom
    :stop calculation geometry:
    emuen file = $EGS_HOME/egs_kerma/emuen_rho_air_1keV-1.5MeV.data    
:stop scoring options:
```
I am using egs_kerma for my study. In fields (A) I need to enter 22363 regions and in (B) 22363 values of equal masses. The only way is typing the values in the input itself?

Thanks. 
-Reginaldo

**Comment by ftessier** [comment]
Looking at the [parsing code for these inputs,](https://github.com/nrc-cnrc/EGSnrc/blob/a6fc389c6465c949645380976f38a013a639759c/HEN_HOUSE/user_codes/egs_kerma/egs_kerma.cpp#L1027-L1032) indeed there is no shorthand notation for long lists of regions or masses. There are two immediate options:

1. Use the linux command `seq -s' '  1 22363` to print the list of regions to terminal, and copy paste in the input file.
2. Modify the source code to allow ranges, for example parse the input `-n1 n2` as meaning all regions from n1 to n2.

**Comment by mainegra** [comment, marked as answer]
@ProfLeao regretfully the input is currently limited to individual entries. Which is indeed shortsighted given that a situation like yours was very likely to occur. I am planning on making the entry of these regions more flexible, similar to the way is done in the egs_dose_scoring ausgab object. Perhaps it would be easy for you to implement that in egs_kerma by looking at how it's done in egs_dose_scoring? It might be a faster and more practical solution than managing thousands of region entries in your input file.

**Reply by ProfLeao** [reply]
Hello @ftessier and @mainegra  I appreciate the contributions. @mainegra, I really chose to follow the way you indicated here and I was successful.
Thank you all.
-Reginaldo

**Reply by mainegra** [reply]
That is great news @ProfLeao! I assume you implemented the multiple group entry scheme from egs_dose_scoring? If yes, and you feel we could all benefit from it, you could contribute that version via a pull request! ;-) Saves us time and helps others! Win-win!

**Reply by ProfLeao** [reply]
```
:start ausgab object definition:
    :start ausgab object:
    library           = egs_dose_scoring
    name              = PDP1
    medium dose = yes
    region dose = yes
        :start output dose file:
            geometry name = EixoCentral1
            file type = 3ddose 
        :stop output dose file:
    :stop ausgab object:
    :start ausgab object:
    library           = egs_dose_scoring
    name              = PDP2
    medium dose = no
    region dose = yes
        :start output dose file:
            geometry name = EixoCentral2
            file type = 3ddose 
        :stop output dose file:
    :stop ausgab object:
    :start ausgab object:
    library           = egs_dose_scoring
    name              = OAR
    medium dose = no
    region dose = yes
        :start output dose file:
            geometry name = MyGeo
            file type = 3ddose 
        :stop output dose file:
    :stop ausgab object:
:stop ausgab object definition:
```

I just did that. How and where could I add a pull request for this simple solution?

**Reply by ftessier** [reply]
I see, you simply used the dose scoring object, which seems to provide what you need. No need for a pull request for that, since this is at the input-file level. What we were alluding to is the modification of the egs_kerma source code to allow region specification as a list or other short hand notation.

**Reply by ProfLeao** [reply]
Forgive me for not fully understanding you. Regarding the source code modification, I can look at it in a couple of weeks. I'll schedule myself.
@ftessier thanks for editing the input transcript.

**Reply by mainegra** [reply]
@ProfLeao I see you moved from calculating `kerma` to calculating `dose`. If your phantom is homogenous or there aren't large density differences. i.e., water/air interfaces, this is not a bad approximation well past the buildup region for an external beam. Perhaps you were really after the quantity `dose`?
My suggested solution was to program into `egs_kerma` the way `egs_dose_scoring` defines groups of scoring regions. I will try to get that done ASAP.
**Note that if for your case the kerma approximation is valid, then egs_kerma is more than 20 times more efficient!**

**Reply by walleludvig** [reply]
Hi @mainegra,

Just wondering if you did end up implementing a grouped methods of scoring regions?

I am using `egs_chamber` in a somewhat unconventional way and are trying to score dose in a vocalised flat panel detector. Since this detector contains A LOT of voxels I do not wish to enter this enourmous list of region number into `cavity regions` under the `calculation geometry` definition but hoping to be able to do something like: select the whole detector as one `cavity region`.

 (I am using `egs_dose_scoring` to output a 3ddose file which already should be containing all that I need.) but I need to select something as the `cavity regions`

**Reply by rtownson** [reply]
@walleludvig I don't think there's a way to group regions in egs_chamber yet. If you really want all these regions listed for `cavity regions`, you will just have to go with the very long list. I usually use a small script to generate these numbers and then copy paste.

**Reply by ftessier** [reply]
@walleludvig you can use labels, but you will still have to specify a long list of region numbers in the geometry definition. The advantage is that you don't have to update the list when the surrounding geometry changes. To create a long list of region numbers, you can resort to the terminal `seq` command.

**Reply by walleludvig** [reply]
Hi @ftessier and @rtownson,

Thanks very much for your replies. I suppose copy pasting in a long list of region number and using the labelling feature is not very problematic at all. Only makes my geometry file rather long. However I did a couple of tests with a very reduced resolution detector element and since I am using `egs_dose_scoring` on my geometry object and reporting the outputs to a `3Ddose` file it does not actually seem to be an issue with not specifying any region numbers in the `calculation geometry` section. Is this correct?

To be clear: I am scoring dose using `egs_dose_scoring` in a `ausgab object` and output a 3Ddose file which I take as input to python to visualise the dose deposition on my flat panel detector. This is actually the only dose that I care about, so to me it seems that the `calculation geometry` in the `scoring options` section does not affect my results in the 3Ddose file. Just wondering if I am correct about this?

**Reply by ftessier** [reply]
Yes, if you are using the dose scoring object, by default it scores in every region, if I remember correctly. The .3ddose file output is completely independent from the calculation geometry cavity regions.
